src/svc_dg.c

```diff
@@ -421,10 +422,11 @@ svc_dg_destroy_task(struct work_pool_entry *wpe)
 		return;
 	}
 
-	if ((rec->xprt.xp_flags & SVC_XPRT_FLAG_CLOSE)
+	xp_flags = atomic_postclear_uint16_t_bits(&rec->xprt.xp_flags,
+						  SVC_XPRT_FLAG_CLOSE);
+	if ((xp_flags & SVC_XPRT_FLAG_CLOSE)
 	    && rec->xprt.xp_fd != RPC_ANYFD) {
 		(void)close(rec->xprt.xp_fd);
-		rec->xprt.xp_fd = RPC_ANYFD;
```

Comment:
Is there a reason to remove the FD poisoning?

Reply:
Is there a reason to keep it? The new postclear SVC_XPRT_FLAG_CLOSE removes any ability to close xp_fd here. This code can only be called once anyway, so it was never important. And it made the full debug log look funky.

Reply:
Because leaking an FD is a big problem in many ways, and 2 instructions in a shutdown path is a small price to pay for a guarantee.  I can imagine wanting to dynamically create and destroy transports in the future, putting this path into the normal runtime.

Reply:
Fine, I'll put them back. As you say, small price to pay.